src/net/download.rs

```diff
@@ -0,0 +1,1904 @@
+                            continue;
+                        },
+                        Err(e) => {
+                            return Err(e.into());
+                        }
+                    };
+
+                    if block_header.parent_microblock == EMPTY_MICROBLOCK_PARENT_HASH && block_header.parent_microblock_sequence == 0 {
+                        // this block doesn't confirm a microblock stream
+                        test_debug!("Block {}/{} does not confirm a microblock stream", &burn_header_hash, &block_hash);
+                        continue;
+                    }
+
+                    // does this anchor block _confirm_ a microblock stream that we don't know about?
+                    let parent_header_opt = {
+                        let mut burn_tx = burndb.tx_begin().map_err(net_error::DBError)?;
+                        match StacksChainState::load_parent_block_header(&mut burn_tx, &chainstate.blocks_db, &chainstate.blocks_path, &burn_header_hash, &block_hash) {
+                            Ok(header_opt) => header_opt,
+                            Err(chainstate_error::DBError(db_error::NotFoundError)) => {
+                                // we don't know about this parent block yet
+                                test_debug!("{:?}: Do not have parent of anchored block {}/{} yet, so cannot ask for the microblocks it produced", &self.local_peer, &burn_header_hash, &block_hash);
+                                continue;
+                            },
+                            Err(e) => {
+                                return Err(e.into());
+                            }
+                        }
+                    };
+
+                    if let Some((parent_header, parent_burn_block_hash)) = parent_header_opt {
+                        if chainstate.get_confirmed_microblock_index_hash(&StacksBlockHeader::make_index_block_hash(&parent_burn_block_hash, &parent_header.block_hash()))?.is_some() {
+                            // we already have this stream, but make sure it connects to _this_ block.  Multiple anchored blocks can descend from the same microblock stream!
+                            if StacksChainState::load_staging_microblock(&chainstate.blocks_db, &parent_burn_block_hash, &parent_header.block_hash(), &block_header.parent_microblock)?.is_some() {
+                                test_debug!("{:?}: Already have microblock stream confirmed by {}/{} (built by {}/{})", &self.local_peer, &burn_header_hash, &block_hash, &parent_burn_block_hash, &parent_header.block_hash());
+                                continue;
+                            }
+                        }
+                        else if StacksChainState::get_microblock_stream_head_hash(&chainstate.blocks_db, &parent_burn_block_hash, &parent_header.block_hash())?.is_some() {
```

Comment:
What's the difference between these two branches (line 720 and 727)? Is there ever a case in which `load_staging_microblock` returns Some and `get_microblock_stream_head_hash` returns None? If not, the whole first branch could just be dropped.

Reply:
The branch at 720 is checking if we have a _processed_ microblock stream that _starts_ with the parent anchored block, and if so, it checks to see if the stream _ends_ with the child anchor block.  If so, then we don't need to fetch this confirmed microblock stream.

The branch at 727 is checking if we have an _unprocessed_ microblock stream that _starts_ with the parent anchor block, and if so, it checks to see if the stream _ends_ with the child anchor block.  If so, then we don't need to fetch this confirmed microblock stream.

Reply:
Yes, but it looks like the condition 727 is a superset of the condition 720, and both branches have the same body.

Reply:
Yes, that's correct.  I added both branches so I could get separate log messages, but I can just drop the first branch.

Reply:
Fixed in 150394c78